crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -407,18 +408,16 @@ impl PubGrubReportFormatter<'_> {
         index_locations: &Option<IndexLocations>,
         unavailable_packages: &FxHashMap<PackageName, UnavailablePackage>,
         incomplete_packages: &FxHashMap<PackageName, BTreeMap<Version, IncompletePackage>>,
+        fork_urls: &ForkUrls,
     ) -> IndexSet<PubGrubHint> {
         let mut hints = IndexSet::default();
         match derivation_tree {
             DerivationTree::External(
                 External::Custom(package, set, _) | External::NoVersions(package, set),
             ) => {
-                if let PubGrubPackageInner::Package {
-                    name, url: None, ..
-                } = &**package
-                {
+                if let PubGrubPackageInner::Package { name, .. } = &**package {
                     // Check for no versions due to pre-release options.
-                    if let Some(selector) = selector {
+                    if let (None, Some(selector)) = (fork_urls.get(name), selector) {
```

Comment:
I personally might do this with `fork_urls.contains_key(name)` rather than `if let None` but subjective :)